src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -95,6 +96,7 @@
          ErrorReport="$(ErrorReport)"
          Features="$(Features)"
          FileAlignment="$(FileAlignment)"
+         GeneratedFilesOutputPath="$(CompilerGeneratedFilesOutputPath)"
```

Comment:
There likely should be an output from the task that includes the written generated files into FileWrites, so later builds delete the previously outputted files and a MSBuild clean would actually delete them. Is there a plan to include that? My impression is the file tracker mechanism you did write tests isn't used for that, but I could be wrong.

Reply:
So this isn't as simple as it sounds; we don't currently return anything like that back from the compiler server, and would need to update the protocol to support returning a list of files written which isn't particularly cheap.

We did speak with the MSBuild team about this, which I had totally forgotten about until now. The idea we came up with was to write out another, known file in the location with a list of everything we wrote. Then we can calculate the FileWrites in the msbuild layer after compilation. I'll implement that now as part of this.

Reply:
I'm going to punt this to another PR as I want to get what we have so far in, and have a good idea how to achieve this. Opened #47966 to track that work.